File: Tutorial_Sample_Code/ReportDocObjectModel/SavedDataSelectionFormula/CS/CS_Win_RDObjMod_SavedDataSelectionFormula/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CS_Win_RDObjMod_SavedDataSelectionFormula
{
    
    public partial class Form1 : Form
    {
        private CustomerBySalesName customerBySalesNameReport;
        private string salesAmount;
        private string operatorValue;
        private string customerName;
        private bool useDefaultValues = true;
        
        public Form1()
        {
            InitializeComponent();
        }

        protected void Page_Unload(object sender, EventArgs e)
        {
            customerBySalesNameReport.Close();
            crystalReportViewer.Dispose();
        }

        private void ConfigureCrystalReports()
        {
            if (useDefaultValues)
            {
                salesAmount = "4000";
                operatorValue = "<";
                customerName = "K";
                operatorValueList.DataSource = System.Enum.GetValues(typeof(CeComparisonOperator));
            }

            string selectionFormula = "{Customer.Last Year's Sales} > " + salesAmount
            + " AND Mid({Customer.Customer Name}, 1, 1) " + operatorValue + "'" + customerName + "'";
            customerBySalesNameReport = new CustomerBySalesName();
            customerBySalesNameReport.DataDefinition.RecordSelectionFormula = selectionFormula;
            crystalReportViewer.ReportSource = customerBySalesNameReport;
            formula.Text = selectionFormula;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ConfigureCrystalReports();
        }

        private void redisplay_Click(object sender, EventArgs e)
        {
            salesAmount = lastYearsSales.Text;
            operatorValue = GetSelectedOperator();
            customerName = letterOfName.Text;
            useDefaultValues = false;

            string selectionFormula = "{Customer.Last Year's Sales} > " + salesAmount
            + " AND Mid({Customer.Customer Name}, 1, 1) " + operatorValue + "'" + customerName + "'";
            customerBySalesNameReport.DataDefinition.SavedDataSelectionFormula = selectionFormula;
            crystalReportViewer.ReportSource = customerBySalesNameReport;
            formula.Text = selectionFormula;
        } 

        private string GetSelectedOperator()
        {
            string selectedOperator = "";

            switch ((CeComparisonOperator)operatorValueList.SelectedIndex)
            {
                case CeComparisonOperator.EqualTo:
                    selectedOperator = "=";
                    break;
                case CeComparisonOperator.GreaterThan:
                    selectedOperator = ">";
                    break;
                case CeComparisonOperator.GreaterThanOrEqualTo:
                    selectedOperator = ">=";
                    break;
                case CeComparisonOperator.LessThan:
                    selectedOperator = "<";
                    break;
                case CeComparisonOperator.LessThanOrEqualTo:
                    selectedOperator = "<=";
                    break;
                case CeComparisonOperator.NotEqualTo:
                    selectedOperator = "<>";
                    break;
            }

            return selectedOperator;
        }
    }
}

File: Tutorial_Sample_Code/ReportDocObjectModel/SavedDataSelectionFormula/VB/VB_Win_RDObjMod_SavedDataSelectionFormula/Form1.vb
Imports CrystalDecisions.CrystalReports.Engine
Imports CrystalDecisions.Shared
Public Class Form1
    Private customerBySalesNameReport As CustomerBySalesName
    Private salesAmount As String
    Private operatorValue As String
    Private customerName As String
    Private useDefaultValues As Boolean = True

    Private Sub Form1_Unload(ByVal sender As System.Object, ByVal e As System.EventArgs)
        customerBySalesNameReport.Close()
        myCrystalReportViewer.Dispose()
    End Sub

    Private Sub ConfigureCrystalReports()
        If useDefaultValues Then
            salesAmount = "4000"
            operatorValue = "<"
            customerName = "K"
            operatorValueList.DataSource = System.Enum.GetValues(GetType(CeComparisonOperator))
        End If

        Dim selectionFormula As String = "{Customer.Last Year's Sales} > " & salesAmount _
        & " AND Mid({Customer.Customer Name}, 1, 1) " & operatorValue & "'" & customerName & "'"
        customerBySalesNameReport = New CustomerBySalesName()
        customerBySalesNameReport.DataDefinition.RecordSelectionFormula = selectionFormula
        myCrystalReportViewer.ReportSource = customerBySalesNameReport
        formula.Text = selectionFormula
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ConfigureCrystalReports()
    End Sub

    Private Sub redisplay_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles redisplay.Click
        salesAmount = lastYearsSales.Text
        operatorValue = GetSelectedOperator()
        customerName = letterOfName.Text
        useDefaultValues = False

        Dim selectionFormula As String = "{Customer.Last Year's Sales} > " & salesAmount _
        & " AND Mid({Customer.Customer Name}, 1, 1) " & operatorValue & "'" & customerName & "'"
        customerBySalesNameReport.DataDefinition.SavedDataSelectionFormula = selectionFormula
        myCrystalReportViewer.ReportSource = customerBySalesNameReport
        formula.Text = selectionFormula
    End Sub

    Private Function GetSelectedOperator() As String
        Dim selectedOperator As String = ""

        Select Case operatorValueList.SelectedIndex
            Case CeComparisonOperator.EqualTo
                selectedOperator = "="
            Case CeComparisonOperator.GreaterThan
                selectedOperator = ">"
            Case CeComparisonOperator.GreaterThanOrEqualTo
                selectedOperator = ">="
            Case CeComparisonOperator.LessThan
                selectedOperator = "<"
            Case CeComparisonOperator.LessThanOrEqualTo
                selectedOperator = "<="
            Case CeComparisonOperator.NotEqualTo
                selectedOperator = "<>"
        End Select

        Return selectedOperator
    End Function
End Class
